packages/mask/src/components/DataSource/usePersonaConnectStatus.ts

```diff
@@ -38,3 +41,34 @@ export function usePersonaConnectStatus() {
         return { connected, action, hasPersona: !!personas.length, currentConnectedPersona }
     }, [personas, lastRecognized, activatedSocialNetworkUI])
 }
+
+export function usePersonaAgainstSNSConnectStatus() {
```

Comment:
This hook looks overcomplicated can you simplify it? And please move it to its own file.

Reply:
Moved, but don't know how to simplify it now.